### DISCUSSION TITLE
smoothly transitioned race bar chart in Plot

### DISCUSSION THREAD

**Discussion by espinielli** [observablehq/plot#2261] [Q&A]
I have an [horizontal bar chart](https://observablehq.com/d/ccb1bf81636664a2) which I can *animate* by updating the relevant dataset.
My goal is to have something similar to [Bar Chart Race](https://observablehq.com/@d3/bar-chart-race) but in Plot.
The current version in Plot is not smoothly transitioning as in the D3 version and the technique in [this notebook](https://observablehq.com/@fil/plot-animate-a-bar-chart/2) is beyond my abilities.
Any suggestions?

**Comment by Fil** [comment]
I've started playing with the dataset, but it appears really "full of holes" — missing data where you need to fill in the blanks. For example RyanAir is ranked number 1 for the 4 dates when it exists, but completely absent on the other dates. Is it that they have 0 traffic, or just that they are ranked > 10? It would be great to have a bit more expert knowledge to understand what happens here.

(The general idea for this "bar chart race" is that we have to be able to model the value (flights) and rank of a given airline as a continuous function of the date — that coincides with the actual values on the actual dates.)

**Reply by espinielli** [reply]
I just took some data points (once per week from a 7-day moving average) from [this dataset](https://observablehq.com/d/96025e5100e0020f)
I didn't want to play with daily ranking because it is too jumpy.

But anyway airlines can enter or exit the top 10...from what you say about the data I find it strange Ryanair jumps out...in reality they should stay close to the top...
